proxy/proxy.go

```diff
@@ -533,6 +539,26 @@ func callWeaveAttach(container *docker.Container, args []string) error {
 	return nil
 }
 
+func validateCIDRs(cidrs []string) error {
+	check := func(cidr string, prefixes []string) bool {
+		for _, prefix := range prefixes {
+			if strings.HasPrefix(cidr, prefix) {
+				if _, err := address.ParseCIDR(strings.TrimPrefix(cidr, prefix)); err == nil {
+					return true
+				}
+			}
+		}
+		return false
+	}
+
+	for _, cidr := range cidrs {
+		if cidr != "net:default" && !check(cidr, []string{"ip:", "net:", ""}) {
```

Comment:
Why special-case "net:default"? What you are checking here has nothing to do with docker's `--net` option.